OpenOversight/app/main/views.py

```diff
@@ -686,62 +684,41 @@ def all_data():
 @limiter.limit('250/minute')
-def upload(department_id):
+def upload(department_id, officer_id=None):
+    if officer_id:
+        officer = Officer.query.filter_by(id=officer_id).first()
+        if not officer:
+            return jsonify(error='This officer does not exist.'), 404
+        if not (current_user.is_administrator or
+                (current_user.is_area_coordinator and officer.department_id == current_user.ac_department_id)):
+            return jsonify(error='You are not authorized to upload photos of this officer.'), 403
     file_to_upload = request.files['file']
     if not allowed_file(file_to_upload.filename):
         return jsonify(error="File type not allowed!"), 415
-    original_filename = secure_filename(file_to_upload.filename)
-    image_data = file_to_upload.read()
-
-    # See if there is a matching photo already in the db
-    hash_img = compute_hash(image_data)
-    hash_found = Image.query.filter_by(hash_img=hash_img).first()
-    if hash_found:
-        return jsonify(error="Image already uploaded to OpenOversight!"), 400
-
-    # Generate new filename
-    file_extension = original_filename.split('.')[-1]
-    new_filename = '{}.{}'.format(hash_img, file_extension)
-
-    # Save temporarily on local filesystem
-    tmpdir = tempfile.mkdtemp()
-    safe_local_path = os.path.join(tmpdir, new_filename)
-    with open(safe_local_path, 'wb') as tmp:
-        tmp.write(image_data)
-    os.umask(SAVED_UMASK)
-
-    # Upload file from local filesystem to S3 bucket and delete locally
-    try:
-        url = upload_file(safe_local_path, original_filename,
-                          new_filename)
-        # Update the database to add the image
-        try:
-            new_image = Image(filepath=url, hash_img=hash_img, is_tagged=False,
-                              date_image_inserted=datetime.datetime.now(),
-                              department_id=department_id,
-                              # TODO: Get the following field from exif data
-                              date_image_taken=datetime.datetime.now(),
-                              user_id=current_user.id)
-        except AttributeError:
-            new_image = Image(filepath=url, hash_img=hash_img, is_tagged=False,
-                              date_image_inserted=datetime.datetime.now(),
-                              department_id=department_id,
-                              # TODO: Get the following field from exif data
-                              date_image_taken=datetime.datetime.now())
-        db.session.add(new_image)
-        db.session.commit()
-        return jsonify(success="Success!"), 200
-    except:  # noqa
-        exception_type, value, full_tback = sys.exc_info()
-        current_app.logger.error('Error uploading to S3: {}'.format(
-            ' '.join([str(exception_type), str(value),
-                      format_exc()])
-        ))
+    image = get_uploaded_image(image=file_to_upload, department_id=department_id)
```

Comment:
`get_uploaded_image()` expects an Image object, but here it's being passed `file_to_uploaded`, which is a `werkzeug.datastructures.FileStorage` object. I think this is what caused the error I ran into, since the passed image doesn't have a `department_id` or `date_image_taken` property (utils.py:482).